### PR TITLE
Add precommit hook

### LINKED ISSUE #13: Move linting and formatting from GitHub Actions to a locally-run pre-commit hook
> I haven't yet set up the pre-commit hook for this project.  Right now the linting/formatting is just being handled by the CI.  We should probably make an issue to set it up though.  
> 
> In my opinion it's a better workflow to catch your errors locally and make any changes on the spot rather than waiting for the CI to fail and having to go back and make a new commit to clean things up.
> 
> This comment is mostly as a note to myself, but if you want, feel free to open an issue to take care of that.

_Originally posted by @aedwardg in https://github.com/CodecademyCommunity/community-bot/pull/11#discussion_r720845397_

Should also update the Advanced Setup section of CONTRIBUTING.md.

### PR DESCRIPTION
**Pull request by Victoria-DR**
## What issue is this solving?

Closes #13 

### Description

Adds a pre-commit hook that lints and formats the code. Same process as with the moderation bot. Now errors can be caught before they are committed.

## Any helpful knowledge/context for the reviewer?

- Is a re-seeding of the database necessary? No
- Any new dependencies to install? Yes
- Any special requirements to test? No
  - (e.g., admin perms, alt account, etc.)

### Please make sure you've attempted to meet the following coding standards
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code

### CODE REVIEW

**Review by aedwardg — APPROVED** [review]
Looks like it should work, but please don't merge until #18 and #20 have been merged and this has been rebased

**Review by mikejoh12 — APPROVED** [review]

**Pull request merged**
